### ISSUE TITLE
ANSI escape sequences does not work with PowerShell 5.1

### ISSUE THREAD

**Issue by fukusuket** [PMunch/nancy#4]
Thank you so much for awesome tool :)
I'll report that I encountered the following issue while trying to integrate Nancy into our [Takajo project](https://github.com/Yamato-Security/takajo)

**Describe the issue**
ANSI color codes no longer work with PowerShell(~5.1)

**Step to Reproduce**
1. Open PowerShell 5.1 console(**not Windows Terminal!**)
2. [Execute README's examlple](https://github.com/PMunch/nancy#nancy---nim-fancy-ansi-tables) 

**Actual behavior**
The text is not colored as shown below.
<img width="960" alt="issue-tmp" src="https://github.com/PMunch/nancy/assets/41001169/5879c023-d74e-40dc-b8a5-1c520a212fd0">

**Expected behavior**
The text is colored.

**Environment**
 - OS: Windows11
 - Nim version 2.0.0
 - nimble v0.14.2 compiled at 2023-08-01 01:51:39

**Additional context**
Nancy using following ANSI escape sequences internally,
https://github.com/PMunch/termstyle/blob/master/termstyle.nim#L23-L46

But PowerShell 5.1 does not support  ANSI escape sequences. (PowerShell 7 or Windows Terminal does support)
https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_ansi_terminals?view=powershell-5.1

If you use the Nim standard functions below, this issue does not seem to occur because the Windows API is executed directly (instead of using ANSI escape sequences).
https://nim-lang.org/docs/terminal.html

**Comment by fukusuket** [comment]
Although it is not a complete fix...  I was able to partially resolve the issue by using [stdout.styledWrite](https://nim-lang.org/docs/terminal.html#styledWrite.m%2CFile%2Cvarargs%5Btyped%5D) as shown below.
 - https://github.com/PMunch/nancy/compare/master...fukusuket:nancy:use-styled-write-insted-of-ansi-escape

Do you have any good ideas to avoid this issue? :)
Thanks,

**Comment by PMunch** [comment]
Hmm, Nancy wasn't really written to work on Windows at all. I wasn't even aware that more recent Windows shells supported ANSI. The input to and output from Nancy in supposed to be ANSI, so I think it would confuse the project if it moved away from that. However Nancy was written to support the implementation of other formats, so you can have a look at the implementation of `echoTable`/`echoTableSeps` to see how these default representations are built and create your own custom one which replaces the ANSI codes with the Windows stuff:
```nim
proc echoTable*(table: TerminalTable, maxSize = terminalWidth(), padding = 1) =
  ## Writes out the table to the terminal with wrapping and styles. `maxSize`
  ## can be used to set the maximum width of the table, this defaults to the
  ## current width of the terminal. `padding` can be used to set how much space
  ## is inserted between each column. If you want to have a table with
  ## separators use `echoTableSeps` instead.
  let sizes = table.getColumnSizes(maxSize, padding = padding)
  for _, entry in table.entries(sizes):
    for _, row in entry():
      for i, cell in row():
        stdout.write cell & (if i != sizes.high: ' '.repeat(padding) else: "")
      stdout.write "\n"
```

**Comment by fukusuket** [comment]
@PMunch 
Thank you so much for fast reply :)

> The input to and output from Nancy in supposed to be ANSI, so I think it would confuse the project if it moved away from that. 

Yes, I feel the same way!  Since PowerShell 5.1 is a special case, I also think that changes for this purpose are not desirable.

> implementation of echoTable/echoTableSeps to see how these default representations are built and create your own

Yes! I just tried the custom implementation below in our Takajo project and it worked.
- https://github.com/Yamato-Security/takajo/pull/57#issuecomment-1778873353
- https://github.com/Yamato-Security/takajo/pull/57/commits/a8ce71f545b6adb2e67f24a5fab14109d2938338

Since the problem has been resolved, I will close the issue.
Thanks again for creating such a great tool :)
